src/scitoken_helper_check.h

```diff
@@ -0,0 +1,24 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_AUTHZ_SCITOKEN_HELPER_CHECK_H_
+#define CVMFS_AUTHZ_SCITOKEN_HELPER_CHECK_H_
+
+#include <cstdio>
+#include <string>
+
+enum StatusSciTokenValidation {
+  kCheckTokenGood,
+  kCheckTokenInvalid,
+};
+
+typedef StatusSciTokenValidation (*CheckSciToken_t)(const std::string &membership,
+                                                    FILE *fp_token);
+
+extern "C" {
+StatusSciTokenValidation CheckSciToken(const std::string &membership,
```

Comment:
Very weird to have a function taking C++ arguments that has C linkage.  I understand why you did it like this, but can you convert the membership argument to `const char *` instead?